autonat_test.go

```diff
@@ -173,6 +188,7 @@ func TestAutoNATPublictoPrivate(t *testing.T) {
 	if status != network.ReachabilityPrivate {
 		t.Fatalf("unexpected NAT status: %d", status)
 	}
+	cancel()
```

Comment:
why?

Reply:
prevents a race where the deferred resetting of `manet.Private4` might be updated while still being accessed by the nat.